core/src/test/scala/org/apache/spark/SparkContextSuite.scala

```diff
@@ -106,6 +106,35 @@ class SparkContextSuite extends FunSuite with LocalSparkContext {
       sc.stop()
     }
   }
+  
+  test("addFile works with relative path") {
```

Comment:
Rather than copy and paste the test I wonder if this can just be injected into the "addFile works" test. Like, make the test file a relative path there. Also I see that test doesn't use the framework provided temp dir which you did correctly here.